.github/workflows/unit_tests.yaml

```diff
@@ -21,7 +21,7 @@ jobs:
           - '2.3'
           - '2.7'
           - '3.0'
-          - '3.1'
+          - '3.2.0-preview2'
```

Comment:
I think the branch protection rules are still requiring 3.1, so those will need to be updated.

Also wondering if we can add 3.1 for Windows, until the `3.2.0-preview2` version becomes available? And update branch protection rules.

Reply:
I've added Ruby 3.1 to the Windows testing matrix and updated the branch protection rules accordingly